lib/skywalking/tracer.lua

```diff
@@ -28,9 +28,7 @@ local nginxComponentId = 6000
 local Tracer = {}
 
 
-function Tracer:start(upstream_name, correlation)
-    local log = ngx.log
-    local WARN = ngx.WARN
+function Tracer:start(upstream_name, correlation, is_propagation)
```

Comment:
Could you explain the new parameter and use case? I am a little confused.

Reply:
It will not set the propagation header to request when the is_propagation is false. 
If the peer is updated, the propagation header will become invalidated. 
So I add this parameter for disabling propagation header when the one want to update they peer.
Surely, I added a method for manual propagation in right moment.

Reply:
Why stop the propogation I mean?
Peer could be updated once header is not set. Even it is set aftet injection, it is the integration level bug breaking the trace.
So, what I am confused is the use case. When and how should we use this parameter.
I assume you have some APISIX integration cases? Could you make it more clear?

Reply:
We can get the actual peer and update the HTTP Header in apisix balance phase(it is also ok for OpenResty).
Because we can set the HTTP Header for outgoing request in the balance phase, it does not raise braking trace.

Reply:
Then you could do an injection later, and change the document about where is the correct phase to inject.
Adding a parameter breaks the API already. If you want, start a tracer and add another step called inject like other agents did. It is better than right now. Propagation is a very official concept in tracing, `is_propagation=false` means breaking context.